gateway-service-knoxtoken/src/main/java/org/apache/knox/gateway/service/knoxtoken/TokenResource.java

```diff
@@ -278,6 +284,39 @@ public Response revoke(String token) {
     return resp;
   }
 
+  @GET
+  @Path(PUBLIC_KEY_PATH)
+  @Produces ({APPLICATION_JSON})
+  public Response publicKey() {
+    GatewayServices services = (GatewayServices) request.getServletContext()
+        .getAttribute(GatewayServices.GATEWAY_SERVICES_ATTRIBUTE);
+
+    KeystoreService ks = services.getService(ServiceType.KEYSTORE_SERVICE);
+
+    try {
+      PublicKey publicKey = ks.getSigningKeystore().getCertificate(getSigningKeyAlias()).getPublicKey();
+      publicKey.getAlgorithm();
+      byte[] pubKeyBytes = publicKey.getEncoded();
+      String base64Key = Base64.encodeBase64String(pubKeyBytes);
```

Comment:
Is this base64 encoded pem format then? Might be worth a separate method that takes the `PublicKey` and describes exactly what is being returned in what format. PEM/DER/base64/etc. 

I don't think it changes the functionality but would make it more clear.

Reply:
@risdenk - could you please elaborate a bit. Are you also suggesting including this information in the response (i.e. will the client need this to interpret the response correctly - that is why I had initially included the algorithm. I'd imagine the format is actually more important).